File: lecture/algorithms/remind/fibonacci/02.js
function fib(index, cache) {
  cache = cache || [];
  if (cache[index]) return cache[index];
  else {
    if (index < 3) return 1;
    else {
      cache[index] = fib(index - 1, cache) + fib(index - 2, cache);
    }
  }
  return cache[index];
}

console.log(fib(20));

File: lecture/algorithms/remind/fibonacci/02.py
def fib(index,cache={}):
    if index in cache.keys():
        return cache[index]
    else:
        if index <3:
            return 1
        else:
            cache[index] = fib(index-1,cache)+fib(index-2,cache)

    return cache[index]

print(fib(50))
